University-of-Auckland-SOFTENG761/supervision
# feat(frontend): connect spectacles to backend

# Description

Links everything to do with spectacles to the backend.
- Spectacles are created on consult creation
- Can see all spectacles on the spectacle details list page.
- To see spectacle details you can either click "SCRIPT" in consult details or go click a row on the spectacles details list page.
- You can view spectacles for a patient in the dispensing records table on the patient details page

## Type of change

# Evidence / Demo

Dispensing records:
![image](https://user-images.githubusercontent.com/68877945/194480562-46c9070c-6727-492e-b225-652a5f44d99e.png)

Spectacle details:
![image](https://user-images.githubusercontent.com/68877945/194480599-ef4a5575-df47-468e-94db-bc38302d8777.png)

Spectacle list:
![image](https://user-images.githubusercontent.com/68877945/194666509-aa577335-b370-4cee-9f13-af6361cc7615.png)


# How Has This Been Tested?

- Clear your cookies and hard reload
- Go to patient details, select a patient
- Click the create new record button, this should create a consult for you along with spectacles
- Fill in some spectacles fields (e.g. spectacles code, colour)
- Click the "SCRIPT" button, and you should be taken to the spectacles details page
- Try changing fields there, check they persist in your database
- Click the spectacles button in the navbar, and see the spectacles list
- Click a row to see if it works
- Go back to the patient details page and view the dispensing records for the patient you just did a consult for
- Click the dispensing record, and check that it takes you to the spectacle details page

**Test Configuration**:
* Browser: Chrome
* Device:  Windows laptop

# Checklist:
